brackets_recur.js

```diff
@@ -0,0 +1,14 @@
+'use strict';
+
+function isEntired(str, index, pair, dict) {
+
+  if (str[index] !== dict[pair]) index = isEntired(str, index+1, str[index], dict);
+
+  return str[index] === dict[pair]
+    ? index : isEntired(str, index+1, pair, dict);
```

Comment:
It's possible to simplify the code here by moving the check for the base of the recursion to the very beginning of this function